File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/EntityVelocityPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record EntityVelocityPacket(int entityId, short x, short y, short z) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        os.writeShort(x);
        os.writeShort(y);
        os.writeShort(z);
    }

    @Override
    public byte getId() {
        return 0x12;
    }

}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/EntityVelocityPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class EntityVelocityPacket(val entityId: Int, val x: Short, val y: Short, val z: Short) : Packet.Outgoing {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId)
        os.writeShort(x, y, z)
    }

    override val id: Byte
        get() = 0x12
}
